src/datachain/lib/file.py

```diff
@@ -206,6 +207,19 @@ def upload(
         file._set_stream(catalog)
         return file
 
+    @classmethod
+    def from_local_path(cls, path: str, catalog: Optional["Catalog"] = None) -> "File":
+        if catalog is None:
+            from datachain.catalog.loader import get_catalog
+
+            catalog = get_catalog()
+
+        parent, name = posixpath.split(path)
+        client = catalog.get_client(parent)
+
+        file_info = client.fs.info(path)
+        return client.info_to_file(file_info, name)
```

Comment:
If you want to be able to call `open` on this `File` you need to `_set_stream` with a catalog.